common/modal.js

```diff
@@ -5,8 +5,11 @@
 
     //TODO
     .factory('modalUtils', ["$uibModal", function ($uibModal) {
-        function showModal(params, successCB, rejectCB) {
+        function showModal(params, successCB, rejectCB, postRenderCB) {
             var modalInstance = $uibModal.open(params);
+            if (postRenderCB) {
+                modalInstance.rendered.then(postRenderCB);    
```

Comment:
Do you know whether there's a case that the promise that this function is returning might get rejected? If you're not sure, or it might get rejected we should add a `catch` block here to just catch that error. I'm not sure what the rejection here might mean. The modal failed to render? Should we rethrow the error? Or should it fail silently? 

When we moved from angular 1.5.x to 1.6.x, we found some places in the code that we forgot to attach a reject callback (or catch block). In these cases, AngularJS will wrap the original error object in another error object and it just makes it harder to debug.

So we should be cautious of adding these behaviors to our code. Unless we're sure that it won't reject the callback.